ci/gpu/build.sh

```diff
@@ -108,11 +108,20 @@ conda config --show-sources
 conda list --show-channel-urls
 
 function install_dask {
-    # Install the main version of dask, distributed, and streamz
-    gpuci_logger "Install the main version of dask, distributed, and streamz"
+    # Install the conda-forge or nightly version of dask and distributed
+    gpuci_logger "Install the conda-forge or nightly version of dask and distributed"
     set -x
-    pip install "git+https://github.com/dask/distributed.git@$DASK_DISTRIBUTED_GIT_TAG" --upgrade --no-deps
-    pip install "git+https://github.com/dask/dask.git@$DASK_DISTRIBUTED_GIT_TAG" --upgrade --no-deps
+    if [[ "${INSTALL_DASK_MAIN}" == 1 ]]; then
+        gpuci_logger "gpuci_mamba_retry update dask"
+        gpuci_mamba_retry update dask
+        gpuci_logger "gpuci_mamba_retry install -c conda-forge dask"
+        gpuci_mamba_retry install -c conda-forge dask
```

Comment:
May want to commit this first to verify that this works:

```suggestion
        gpuci_logger "gpuci_mamba_retry install conda-forge::dask>=2021.11.1 --force-reinstall"
        gpuci_mamba_retry install conda-forge::dask>=2021.11.1 --force-reinstall
```